### PR TITLE
fix: adjust MemberTaskAdminModal condition

### PR DESCRIPTION
**Pull request by LoisChen68**

### CODE REVIEW

**Comment by LoisChen68** [comment]
[(2)【待辦清單】只是想改待辦狀態卻無法修改，跳出此時段無可用zoom帳號+調整同一人時段重複的判斷條件](https://trello.com/c/zyNq6NBA/5332-2%E3%80%90%E5%BE%85%E8%BE%A6%E6%B8%85%E5%96%AE%E3%80%91%E5%8F%AA%E6%98%AF%E6%83%B3%E6%94%B9%E5%BE%85%E8%BE%A6%E7%8B%80%E6%85%8B%E5%8D%BB%E7%84%A1%E6%B3%95%E4%BF%AE%E6%94%B9%EF%BC%8C%E8%B7%B3%E5%87%BA%E6%AD%A4%E6%99%82%E6%AE%B5%E7%84%A1%E5%8F%AF%E7%94%A8zoom%E5%B8%B3%E8%99%9F%E8%AA%BF%E6%95%B4%E5%90%8C%E4%B8%80%E4%BA%BA%E6%99%82%E6%AE%B5%E9%87%8D%E8%A4%87%E7%9A%84%E5%88%A4%E6%96%B7%E6%A2%9D%E4%BB%B6)

File: src/components/task/MemberTaskAdminModal.tsx [hunk 1]
```diff
@@ -208,8 +222,15 @@ const MemberTaskAdminModal: React.FC<
...
-            toBeUsedServiceId = availableZoomServiceId
+            if (
+              memberTaskExecutorId !== formExecutorId &&
+              (memberTaskDueAtTime === formDueAtTime || memberTaskMeetingHours === formMeetingHours)
+            ) {
+              toBeUsedServiceId = overlapMeets.find(meet => meet.id === memberTaskMeetId)?.serviceId || null
+            } else {
+              const availableZoomServiceId = await zoomServiceCheck({ overlapMeets })
+              toBeUsedServiceId = availableZoomServiceId
+            }
```
Comment (Ponchimeow) [inline src/components/task/MemberTaskAdminModal.tsx:233]: could move `deleteMeet` before `const { overlapMeets } = await getOverlapMeets`
then the condition line:225 to line:229 could be remove

File: src/components/task/MemberTaskAdminModal.tsx [hunk 2]
```diff
@@ -185,10 +185,24 @@ const MemberTaskAdminModal: React.FC<
...
         let updatedMeetId = memberTask?.meet?.id || null
 
-        if (hasMeeting) {
+        if (
+          hasMeeting &&
+          (memberTaskDueAtTime !== formDueAtTime ||
+            memberTaskExecutorId !== formExecutorId ||
+            memberTaskMemberId !== formMemberId ||
+            memberTaskMeetingGateway !== formMeetingGateway ||
+            memberTaskMeetingHours !== formMeetingHours)
```
Comment (Ponchimeow) [inline src/components/task/MemberTaskAdminModal.tsx:204]: these condition are unnecessary.

There are two situations where a meeting will be deleted:

1. When there is no meeting.
2. When there is a meeting and meeting-related information has been modified.

File: src/components/task/MemberTaskAdminModal.tsx [hunk 3]
```diff
@@ -286,7 +306,14 @@ const MemberTaskAdminModal: React.FC<
             },
```
Comment (Ponchimeow) [inline src/components/task/MemberTaskAdminModal.tsx:306]: Based on the previous conversation, adjust the conditions and positions accordingly.

**Review by Ponchimeow — CHANGES_REQUESTED** [review]

File: src/components/task/MemberTaskAdminModal.tsx (hunk 1 above)
Comment (LoisChen68) [reply src/components/task/MemberTaskAdminModal.tsx:233]: Thanks to review, it has been adjusted and refactored.

File: src/components/task/MemberTaskAdminModal.tsx (hunk 2 above)
Comment (LoisChen68) [reply src/components/task/MemberTaskAdminModal.tsx:204]: Thanks to review, it has been adjusted and refactored.

File: src/components/task/MemberTaskAdminModal.tsx (hunk 3 above)
Comment (LoisChen68) [reply src/components/task/MemberTaskAdminModal.tsx:306]: Thanks to review, it has been adjusted and refactored.

File: src/types/member.ts [hunk 4]
```diff
@@ -198,6 +199,22 @@ export type MemberTaskProps = {
   } | null
 }
 
+export type MemberTaskAdminModalFieldProps = {
```
Comment (Ponchimeow) [inline src/types/member.ts:202]: prefer naming MemberTaskAdminModalField

**Review by Ponchimeow — APPROVED** [review]

**Pull request merged**